internal/services/elastic/elastic.go

```diff
@@ -0,0 +1,232 @@
+// Package elastic is responsible for handling queries to the backing elastic database.
+package elastic
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"slices"
+	"strconv"
+	"time"
+
+	"github.com/DIMO-Network/device-data-api/internal/config"
+	"github.com/DIMO-Network/device-data-api/internal/constants"
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/typedapi/core/search"
+	"github.com/elastic/go-elasticsearch/v8/typedapi/types"
+	"github.com/rs/zerolog"
+	"github.com/tidwall/gjson"
+)
+
+const (
+	docByInterval  = "documents_by_interval"
+	singleDoc      = "select_single_doc"
+	timeField      = "time"
+	subjectField   = "subject"
+	defaultRetries = 5
+)
+
+// Service is a service for performing queries on elastic.
+type Service struct {
+	settings *config.Settings
+	log      *zerolog.Logger
+	esClient *elasticsearch.TypedClient
+}
+
+// New creates a newly configured elastic service.
+func New(settings *config.Settings, logger *zerolog.Logger, caCert []byte) (*Service, error) {
+	esConfig := elasticsearch.Config{
+		Addresses:  []string{settings.ElasticSearchAnalyticsHost},
+		Username:   settings.ElasticSearchAnalyticsUsername,
+		Password:   settings.ElasticSearchAnalyticsPassword,
+		CACert:     caCert,
+		MaxRetries: defaultRetries,
+	}
+	es8Client, err := elasticsearch.NewTypedClient(esConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create elastic client: %w", err)
+	}
+	eClient := Service{
+		esClient: es8Client,
+		log:      logger,
+		settings: settings,
+	}
+	return &eClient, nil
+}
+
+// EsClient returns the underlying elastic client used by the service.
+//
+// This function exists to maintain previous behavior for endpoint who are not yet using the elastic service.
+func (s *Service) EsClient() *elasticsearch.TypedClient {
+	return s.esClient
+}
+
+// GetHistoryParams defines the parameters for retrieving history from elastic.
+type GetHistoryParams struct {
+	// StartTime is the start time of the history.
```

Comment:
Could we mention the defaults behavior somewhere? Maybe here?

Reply:
Can do. On that note, Do you think the default behavior should be an explicit method of this structure? 
`params.SetDefaults()` that way the caller can choose if they want to use them or not.

Reply:
That's a good question.

Reply:
I like this idea! I believe the params used in the history endpoint are used elsewhere.. do we want to have a more generic params struct? (realizing the defaults may be different elsewhere though so I can see the case for having a specific history one)

Reply:
Where else?

Reply:
ah, nvm-- I wasnt looking at this closely enough I was thinking of http params elsewhere

Reply:
A method seems cool.

Reply:
It sounds like we are not doing this.